ipam/allocator.go

```diff
@@ -126,12 +126,14 @@ func (alloc *Allocator) cancelOps(ops *[]operation) {
 // if we found any.
 func (alloc *Allocator) cancelOpsFor(ops *[]operation, ident string) bool {
 	var found bool
-	for i, op := range *ops {
-		if op.ForContainer(ident) {
+	for i := 0; i < len(*ops); {
+		if op := (*ops)[i]; op.ForContainer(ident) {
 			found = true
 			op.Cancel()
 			*ops = append((*ops)[:i], (*ops)[i+1:]...)
+			continue
 		}
+		i++
```

Comment:
The relevant part is:

> The range expression is evaluated once before beginning the loop

The failure symptom is that `(*ops)[i+1:]` is invalid because `i` is now equal to the length of the array, which has been shortened.